core/modules/css/CssAccess.cc

```diff
@@ -0,0 +1,848 @@
+        throw KeyValueError(pKey, "one of the keys is not numeric: " + std::string(exc.what()));
+    }
+
+    return striping;
+}
+
+void
+CssAccess::createDb(std::string const& dbName,
+                    StripingParams const& striping,
+                    std::string const& storageClass,
+                    std::string const& releaseStatus) {
+    LOGF(_log, LOG_LVL_DEBUG, "createDb(%1%)" % dbName);
+
+    std::string partId;
+    if (striping.stripes > 0) {
+        // need to define partitioning structure
+
+        // create unique key for it and get its id
+        std::string const pfxKey = _prefix + "/PARTITIONING/_";
+        std::string const partKey = _kvI->create(pfxKey, "", true);
+        partId = partKey.substr(pfxKey.size());
+
+        // store striping structure
+        // TODO: add uuid if we want to use it
+        std::map<std::string, std::string> stripMap{
+            std::make_pair("nStripes", std::to_string(striping.stripes)),
+            std::make_pair("nSubStripes", std::to_string(striping.subStripes)),
+            std::make_pair("overlap", std::to_string(striping.overlap))};
+        _storePacked(partKey, stripMap);
+    }
+
+    // TODO: add uuid if we want to use it
+    std::map<std::string, std::string> dbMap{
+        std::make_pair("releaseStatus", releaseStatus),
+        std::make_pair("storageClass", storageClass)};
+    if (not partId.empty()) {
+        dbMap.insert(std::make_pair("partitioningId", partId));
+    }
+
+    std::string const dbKey = _prefix + "/DBS/" + dbName;
+    _storePacked(dbKey, dbMap);
+    _kvI->set(dbKey, "READY");
+}
+
+void
+CssAccess::createDbLike(std::string const& dbName,
+                        std::string const& templateDbName) {
+    LOGF(_log, LOG_LVL_DEBUG, "createDbLike(%1%)" % dbName);
+
+    std::vector<std::string> subKeys{"partitioningId", "releaseStatus", "storageClass"};
+    auto dbMap = _getSubkeys(_prefix + "/DBS/" + templateDbName, subKeys);
+    if (dbMap.empty()) {
+        // nothing is found, check whether db exists
+        _assertDbExists(templateDbName);
+    }
+
+    // make new database with the copy of all parameters
+    std::string const dbKey = _prefix + "/DBS/" + dbName;
+    _storePacked(dbKey, dbMap);
+    _kvI->set(dbKey, "READY");
+}
+
+void
+CssAccess::dropDb(std::string const& dbName) {
+    LOGF(_log, LOG_LVL_DEBUG, "dropDb(%1%)" % dbName);
+
+    std::string key = _prefix + "/DBS/" + dbName;
+
+    // key is supposed to exist and key.json is optional
+    try {
+        LOGF(_log, LOG_LVL_DEBUG, "dropDb: try to delete packed key: %s.json" % key);
+        _kvI->deleteKey(key + ".json");
+    } catch (NoSuchKey const& exc) {
+        LOGF(_log, LOG_LVL_DEBUG, "dropDb: packed key is not found");
+    }
+    try {
+        LOGF(_log, LOG_LVL_DEBUG, "dropDb: try to delete regular key: %s" % key);
+        _kvI->deleteKey(key);
+    } catch (NoSuchKey const& exc) {
+        LOGF(_log, LOG_LVL_DEBUG, "dropDb: regular key is not found");
```

Comment:
I'm wondering if this should be info level. If so, it should probably include the key name.